### PR TITLE
Clear Yes Nos Part 1: Integrations testing in their own database

### PR DESCRIPTION
**Pull request by macrael**
## Description

On kickback, clear the yes/nos that need to be resubmitted by the applicant. 

This PR also includes integration tests and a dbmigrate binary to setup the test db. 

The task for clearing yes nos is quite large so I am breaking it up into some chunks. 

## Checklist for Reviewer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

### CODE REVIEW

**Comment by macrael** [comment]
Addresses https://github.com/18F/e-QIP-prototype/issues/738

File: api/cmd/dbmigrate/main.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+			}
+
+			fmt.Println(text)
+			if !(text == "y" || text == "Y" || text == "YES" || text == "yes") {
+				return errors.New("user disconfirmed reset")
+			}
+
+		}
+
+		dropCmd := "DROP DATABASE " + dbName
```
[the remark this answers is not in the document]
Comment (macrael) [reply api/cmd/dbmigrate/main.go]: Unfortunately, I don't think that placeholder parameters work for DROP DATABASE. I tried that and was told that it is the one command where you can't use them.

[the remark this answers is not in the document]
Comment (macrael) [reply api/cmd/dbmigrate/main.go]: I would love to be wrong though

File: api/integration/sections/history.go [hunk 2]
```diff
@@ -0,0 +1,732 @@
+package sections
+
+// HistResidenceSingle contains a single residence
+const HistResidenceSingle = `
```
[the remark this answers is not in the document]
Comment (macrael) [reply api/integration/sections/history.go]: This is a good idea. I'll fix that here.

File: api/cmd/dbmigrate/main.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	}
+
+	if subject == "" {
+		return ""
+	}
+
+	return pathToPattern(dir, pattern)
+}
+
+func guessMigrationsPath() string {
```
[the remark this answers is not in the document]
Comment (macrael) [reply api/cmd/dbmigrate/main.go]: How about we specify the path to the migrations directory? I think if we want to specify individual files then we may have to rework the migration library we are using.

File: api/cmd/dbmigrate/main.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
+package main
```
[the remark this answers is not in the document]
Comment (macrael) [reply api/cmd/dbmigrate/main.go]: I think I'll leave that for when we do 1754.

File: api/cmd/dbmigrate/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (macrael) [reply api/cmd/dbmigrate/main.go]: And I was nervous about adding the drop command, I think splitting that out into a separate command is reasonable. I can patch that here, too.

[the remark this answers is not in the document]
Comment (macrael) [reply api/cmd/dbmigrate/main.go]: Went ahead and split it out.

**Comment by macrael** [comment]
@ryanhof I pushed a couple commits that should resolve all your comments. Let me know if I missed anything. Here are the things you should look at specifically: 
1. I split out `dbreset` from `dbmigrate` and made `dbmigrate` require that you specify the path to the migrations directory. If we need to specify the path to specific migration files in the future that will require more work
2. I broke out the test fixtures into json files in the testdata directory, I'll keep migrating things in there to the new directory structure as I get to them in later PRs. 
3. I didn't look at faker too hard, random email usernames was enough to make sure I got new data every time.

File: api/cmd/dbmigrate/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanhofdotgov) [reply api/cmd/dbmigrate/main.go]: Yeah, meant an explicit path to the migrations directory so we are on the same page 👍

**Pull request merged**